ts/Manager/App.tsx

```diff
@@ -121,6 +126,23 @@ const App: React.FC<Props> = () => {
 		});
 	}
 
+	function cloneRoom(room: Room) {
+
+		if (room.moderatorToken !== null) {
+			room.moderatorToken = 'true';
+		}
+
+		return api.createRoom(room.name, room.access, room.maxParticipants).then(newRoom => {
+			api.getRoomShares(room.id).then(shares => shares.forEach(share => {
+				api.createRoomShare(newRoom.id, share.shareType, share.shareWith, share.permission);
+			}))
+
+			room.uid = newRoom.uid;
```

Comment:
We should probably clone the object, instead of modifying it. Those 3 lines could be written as

```
updateRoom({
  ...room,
  uid: newRoom.uid,
  id: newRoom.id,
})
```

Reply:
Ok I see that it's already cloned in `ts/Manager/RoomRow.tsx`, but for a better DX it would be nice to do it here.